extensions/amp-render/1.0/amp-render.js

```diff
@@ -41,7 +41,66 @@ const isAmpStateSrc = (src) => src && src.startsWith(AMP_STATE_URI_SCHEME);
  */
 const isAmpScriptSrc = (src) => src && src.startsWith(AMP_SCRIPT_URI_SCHEME);
 
-class AmpRender extends BaseElement {
+/**
+ * Gets the json from an "amp-state:" uri. For example, src="amp-state:json.path".
+ *
+ * TODO: this implementation is identical to one in amp-list. Move it
+ * to a common file and import it.
+ *
+ * TODO: Add src as a param to this method
+ * (https://github.com/ampproject/amphtml/pull/33189#discussion_r598743971).
+ *
+ * @param {!AmpElement} element
+ * @return {Promise<!JsonObject>}
+ */
+const getAmpStateJson = (element) => {
+  const src = element.getAttribute('src');
+  return Services.bindForDocOrNull(element)
+    .then((bind) => {
+      userAssert(bind, '"amp-state:" URLs require amp-bind to be installed.');
+      const ampStatePath = src.slice(AMP_STATE_URI_SCHEME.length);
+      return bind.getStateAsync(ampStatePath).catch((err) => {
+        const stateKey = ampStatePath.split('.')[0];
+        user().error(
+          TAG,
+          `'amp-state' element with id '${stateKey}' was not found.`
+        );
+        throw err;
+      });
+    })
+    .then((json) => {
+      userAssert(
+        typeof json !== 'undefined',
+        `[amp-render] No data was found at provided uri: ${src}`
+      );
+      return json;
+    });
+};
+
+/**
+ * Returns the correct fetch function for amp-state, amp-script or
+ * to fetch remote JSON.
+ *
+ * @param {!AmpElement} element
+ * @return {Function}
+ */
+export const getJsonFn = (element) => {
+  const src = element.getAttribute('src');
+  if (!src) {
+    // TODO(dmanek): assert that src is provided instead of silently failing below.
+    return () => {};
+  }
+  if (isAmpStateSrc(src)) {
+    return getAmpStateJson.bind(null, element);
```

Comment:
It's not clear from `bind` what the call-signature of the returned function should be. Is this a closure? Is it curried? Prefer using arrows:

```suggestion
    return (src) => getAmpStateJson(element, src);
```

Reply:
Good point, much cleaner using arrow 👍🏽